proxy/src/config.rs

```diff
@@ -135,6 +141,8 @@ pub const ENV_METRICS_LISTENER: &str = "CONDUIT_PROXY_METRICS_LISTENER";
 const ENV_PRIVATE_CONNECT_TIMEOUT: &str = "CONDUIT_PROXY_PRIVATE_CONNECT_TIMEOUT";
 const ENV_PUBLIC_CONNECT_TIMEOUT: &str = "CONDUIT_PROXY_PUBLIC_CONNECT_TIMEOUT";
 pub const ENV_BIND_TIMEOUT: &str = "CONDUIT_PROXY_BIND_TIMEOUT";
+pub const ENV_PRIVATE_PORTS_DISABLE_PROTOCOL_DETECTION: &str = "CONDUIT_PROXY_PRIVATE_PORTS_DISABLE_PROTOCOL_DETECTION";
+pub const ENV_PUBLIC_PORTS_DISABLE_PROTOCOL_DETECTION: &str = "CONDUIT_PROXY_PUBLIC_PORTS_DISABLE_PROTOCOL_DETECTION";
```

Comment:
I think these should be s/PRIVATE/OUTGOING/ and s/PUBLIC/INCOMING/.

"private" and "public" are not user-facing concepts they are part of the internals of Conduit. When we write the documentation for this feature we will (presumably) document one flag as controlling protocol detection for "incoming traffic destined for the given ports" and "outgoing traffic destined for the given ports" and we should use that terminology for these settings.

Reply:
Oh really? Have I misunderstood? I thought "private" config options configured our `Outbound` listener, and "public" configured our `Inbound` listener.

Reply:
My understanding of the two terms is the opposite: PUBLIC refers to the listener that is listening publicly, that anything else can connect to, in order to talk to the app (so, our `Inbound` stack), and PRIVATE refers to the listener that only the app can talk to (so, our `Outbound` stack).

Reply:
Maybe we should talk face-to-face about it since I think we have some miscommunication.

To backtrack a bit, you said earlier:
> I thought "private" config options configured our Outbound listener, and "public" configured our Inbound listener

However, that's not exactly the case. The `Inbound` stack is configured using `ENV_PUBLIC_LISTENER`, `ENV_PRIVATE_FORWARD`, and `ENV_PRIVATE_CONNECT_TIMEOUT`:

```rust
        let inbound_listener = BoundPort::new(config.public_listener.addr)
            .expect("public listener bind");

[...]

        let inbound = {
            [...]
            let default_addr = config.private_forward.map(|a| a.into());
            let fut = serve(
                config.private_connect_timeout,
                Inbound::new(default_addr, bind),
                [...])
            [...]
        };
```

Similarly, the `Outbound` stack is configured using `ENV_PRIVATE_LISTENER` and `ENV_PUBLIC_CONNECT_TIMEOUT`.

One way of thinking about it is that the `Inbound` stack is a pair (public listener, private destination) and the `Outbound` stack is a pair (private listener, public destinations). The new settings you are adding configure the behavior for the "public listener" and the "public destinations"; i.e. the "public" end of both stacks. That's why I said "private" vs. "public" is the wrong distinction: they're both "public". The distinction that matters is that one setting is for the public end of the `Inbound` stack and the other setting is for the public end of the `Outbound` stack.

Reply:
I thought of a new way of explaining the naming rationale: `conduit inject` has these arguments:

```
--skip-inbound-ports uintSlice    ports that should skip the proxy and send directly to the application (default [])
--skip-outbound-ports uintSlice   outbound ports that should skip the proxy (default [])
```

proxy-init has these arguments:

```
--inbound-ports-to-ignore intSlice  
--incoming-proxy-port int             Port to redirect incoming traffic (default -1)
      --outbound-ports-to-ignore intSlice   Outbound ports to ignore and not redirect to proxy. This has higher precedence than any other parameters.
--outgoing-proxy-port
```

So "inbound" and "outbound" are the terms we're already using for the vary similar feature, and those are the terms that we'll almost definitely use in the `conduit inject` arguments that we'll add later. (I think it's odd that proxy-init uses both "inbound/outbound" & "incoming/outgoing".)

I think this is a strong point in favor of using the "inbound" and "outbound" terminology now.